src/js/media-controller.js

```diff
@@ -682,9 +693,17 @@ const getCurrentTime = (controller) => {
 };
 
 const getDuration = (controller) => {
-  const media = controller.media;
+  const media = controller?.media;
+  if (!Number.isFinite(media?.duration)) return NaN;
```

Comment:
is this to handle `Infinity`?

Reply:
yup, `Infinity` and any non-numerics all in one fell swoop.